### DISCUSSION TITLE
Can I run external binary?

### DISCUSSION THREAD

**Discussion by linonetwo** [WasmEdge/WasmEdge#2386] [Q&A] [answered]
I hope I can run https://github.com/antimatter15/alpaca.cpp 's binary from a wasm module, can I achieve this with wasmedge? So I can do C++ speeduped infer at the edge.

**Comment by hydai** [comment, marked as answer]
Is this a library or an external binary tool?
You can write a new plugin to leverage the library or call the `exec`-like functions to run the external binaries.
There is no way to do this unless you use the plugin system or host functions.

**Reply by linonetwo** [reply]
It is a c program that complies to a binary executable, and can load Alpaca language model from filesystem, and receive input from cli.

What do you mean by `exec-like functions`? Is that something like nodejs `child_process.exec()` in your API?

> use the plugin system or host functions

So I can also bind C code to a wasmedge plugin, so my wasmedge instance can infer llama and alpaca model. This might be a hard but normal way.

**Reply by hydai** [reply]
Something like `execvp()` in C/C++.
https://www.digitalocean.com/community/tutorials/execvp-function-c-plus-plus
